service/src/main/kotlin/fi/espoo/evaka/specialdiet/SpecialDietQueries.kt

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.specialdiet
+
+import fi.espoo.evaka.shared.db.Database
+import org.jdbi.v3.core.mapper.PropagateNull
+
+data class SpecialDiet(@PropagateNull val id: Int?, val name: String, val abbreviation: String)
+
+fun Database.Transaction.setSpecialDiets(specialDietList: List<SpecialDiet>) {
+    val newSpecialDietIds = specialDietList.map { it.id }
+    execute {
+        sql("UPDATE child SET diet_id = null WHERE diet_id != ALL (${bind(newSpecialDietIds)})")
+    }
```

Comment:
Fine for now, but for a longer term plan this could be unsafe with severe results if some children suddenly no longer have any diet and no one will notice, even if jamix removing them is unlikely. Not sure how to approach that. Might need to set some diet_missing flag which can then be discovered and handled by users through some report / mega warning. Should be discussed with UX/POs.